### ISSUE TITLE
Error on new install

### ISSUE THREAD

**Issue by jimkd1yv** [phiresky/nmap-log-parse#11]
Got the data collection running under root, confirmed it is producing output files.  Started server with python3 -m http.server.  

When I go to the page, it throws this error:
Error: TypeError: this.props.deviceInfos.get(...) is undefined.
See F12 console for more details.

Here is the console
```
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-23.xml [HTTP/1.0 404 File not found 26ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-22.xml [HTTP/1.0 404 File not found 37ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-21.xml [HTTP/1.0 404 File not found 27ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-20.xml [HTTP/1.0 404 File not found 36ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-19.xml [HTTP/1.0 404 File not found 31ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-18.xml [HTTP/1.0 404 File not found 34ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-17.xml [HTTP/1.0 404 File not found 36ms]
GET 
XHR 
http://192.168.1.128:8000/logs/oldlogs [HTTP/1.0 404 File not found 43ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-23.xml [HTTP/1.0 404 File not found 86ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-22.xml [HTTP/1.0 404 File not found 29ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-21.xml [HTTP/1.0 404 File not found 45ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-23.xml [HTTP/1.0 404 File not found 28ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-22.xml [HTTP/1.0 404 File not found 33ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-21.xml [HTTP/1.0 404 File not found 29ms]
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  bundle.js:4391:10
getDates: timer startedbundle.js:114
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-23.xml [HTTP/1.0 404 File not found 28ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-22.xml [HTTP/1.0 404 File not found 23ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-21.xml [HTTP/1.0 404 File not found 20ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-20.xml [HTTP/1.0 404 File not found 22ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-19.xml [HTTP/1.0 404 File not found 25ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-18.xml [HTTP/1.0 404 File not found 23ms]
GET 
XHR 
http://192.168.1.128:8000/nmap/logs/2017-08-17.xml [HTTP/1.0 404 File not found 31ms]
getDates: 554.22msbundle.js:116
getStatic: timer startedbundle.js:117
getStatic: 2.84msbundle.js:120
TypeError: this.props.deviceInfos.get(...) is undefined
Stack trace:
render@http://192.168.1.128:8000/bin/bundle.js:26246:25
_renderValidatedComponentWithoutOwnerOrContext@http://192.168.1.128:8000/bin/bundle.js:15218:30
_renderValidatedComponent@http://192.168.1.128:8000/bin/bundle.js:15244:28
performInitialMount@http://192.168.1.128:8000/bin/bundle.js:14754:26
mountComponent@http://192.168.1.128:8000/bin/bundle.js:14647:17
mountComponent@http://192.168.1.128:8000/bin/bundle.js:7753:19
performInitialMount@http://192.168.1.128:8000/bin/bundle.js:14760:19
mountComponent@http://192.168.1.128:8000/bin/bundle.js:14647:17
mountComponent@http://192.168.1.128:8000/bin/bundle.js:7753:19
mountComponentIntoNode@http://192.168.1.128:8000/bin/bundle.js:19745:17
perform@http://192.168.1.128:8000/bin/bundle.js:8201:14
batchedMountComponentIntoNode@http://192.168.1.128:8000/bin/bundle.js:19766:4
perform@http://192.168.1.128:8000/bin/bundle.js:8201:14
batchedUpdates@http://192.168.1.128:8000/bin/bundle.js:16855:8
batchedUpdates@http://192.168.1.128:8000/bin/bundle.js:7403:4
_renderNewRootComponent@http://192.168.1.128:8000/bin/bundle.js:19929:6
_renderSubtreeIntoContainer@http://192.168.1.128:8000/bin/bundle.js:20008:22
render@http://192.168.1.128:8000/bin/bundle.js:20029:13
run/<@http://192.168.1.128:8000/bin/bundle.js:123:10
fulfilled@http://192.168.1.128:8000/bin/bundle.js:60:23
callListener@http://192.168.1.128:8000/bin/bundle.js:22068:23
endMicroTickScope@http://192.168.1.128:8000/bin/bundle.js:22157:21
wrap/<@http://192.168.1.128:8000/bin/bundle.js:22243:34
  bundle.js:128:6

And the server log
```
192.168.1.107 - - [23/Aug/2017 15:07:51] "GET / HTTP/1.1" 200 -
192.168.1.107 - - [23/Aug/2017 15:07:51] "GET /bin/bundle.js HTTP/1.1" 200 -
192.168.1.107 - - [23/Aug/2017 15:07:52] "GET /bin/bundle.js.map HTTP/1.1" 200 -
192.168.1.107 - - [23/Aug/2017 15:07:53] "GET /config.json HTTP/1.1" 200 -
192.168.1.107 - - [23/Aug/2017 15:07:53] code 404, message File not found
192.168.1.107 - - [23/Aug/2017 15:07:53] "GET /nmap/logs/2017-08-23.xml HTTP/1.1" 404 -
192.168.1.107 - - [23/Aug/2017 15:07:53] code 404, message File not found
192.168.1.107 - - [23/Aug/2017 15:07:53] "GET /nmap/logs/2017-08-22.xml HTTP/1.1" 404 -
192.168.1.107 - - [23/Aug/2017 15:07:53] code 404, message File not found
192.168.1.107 - - [23/Aug/2017 15:07:53] "GET /nmap/logs/2017-08-21.xml HTTP/1.1" 404 -
192.168.1.107 - - [23/Aug/2017 16:39:16] "GET / HTTP/1.1" 200 -
192.168.1.107 - - [23/Aug/2017 16:39:16] "GET /bin/bundle.js HTTP/1.1" 200 -
192.168.1.107 - - [23/Aug/2017 16:39:17] "GET /bin/bundle.js.map HTTP/1.1" 200 -
192.168.1.107 - - [23/Aug/2017 16:39:17] "GET /config.json HTTP/1.1" 200 -
192.168.1.107 - - [23/Aug/2017 16:39:18] code 404, message File not found
192.168.1.107 - - [23/Aug/2017 16:39:18] "GET /nmap/logs/2017-08-23.xml HTTP/1.1" 404 -
192.168.1.107 - - [23/Aug/2017 16:39:18] code 404, message File not found
192.168.1.107 - - [23/Aug/2017 16:39:18] "GET /nmap/logs/2017-08-22.xml HTTP/1.1" 404 -
192.168.1.107 - - [23/Aug/2017 16:39:18] code 404, message File not found
192.168.1.107 - - [23/Aug/2017 16:39:18] "GET /nmap/logs/2017-08-21.xml HTTP/1.1" 404 -

```

```

**Comment by phiresky** [comment]
From what I can tell from your logs, the client did not find a single xml log file, correct? The given failure looks like it could result from simply not having any log data available. Did you make sure the log file path is correct? (`/nmap/logs/2017-08-21.xml`)

**Comment by jimkd1yv** [comment]
Logs are most definitely there.  I think the code is looking in the wrong place.  How do I change the code to look under /var/www/nmap-logs ?
```
root@gdoor:/home/pi/applications/nmap-log-parse-master# ls -l /var/www/nmap-logs/
total 120
-rw-r--r-- 1 root root 114854 Aug 23 23:00 2017-08-23.xml

root@gdoor:/home/pi/applications/nmap-log-parse-master# head  /var/www/nmap-logs/2017-08-23.xml 
<?xml version="1.0"?>
<!DOCTYPE nmaprun>
<?xml-stylesheet href="file:///usr/bin/../share/nmap/nmap.xsl" type="text/xsl"?>
<!-- Nmap 6.47 scan initiated Wed Aug 23 14:30:01 2017 as: nmap -sn -oX - 192.168.1.* -->
<nmaprun scanner="nmap" args="nmap -sn -oX - 192.168.1.*" start="1503513001" startstr="Wed Aug 23 14:30:01 2017" version="6.47" xmloutputversion="1.04">
<verbose level="0"/>
<debugging level="0"/>
<host><status state="up" reason="arp-response" reason_ttl="0"/>
<address addr="192.168.1.1" addrtype="ipv4"/>
<address addr="48:F8:B3:**:**:**" addrtype="mac" vendor="Cisco-Linksys"/>
root@gdoor:/home/pi/applications/nmap-log-parse-master# 

```

**Comment by jimkd1yv** [comment]
I tried moving everything that I pulled into the same directory where the log files are, i.e., /var/www/nmap-logs/ and then I updated the config.json to point to same.  The error shown when I fetch the page is the same.  Note that the http server log says the page is not found, even though I look and see the log files are definitely there:

```
root@gdoor:/var/www/nmap-logs# 192.168.1.107 - - [24/Aug/2017 08:37:04] "GET / HTTP/1.1" 200 -
192.168.1.107 - - [24/Aug/2017 08:37:04] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:04] "GET /favicon.ico HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:04] "GET /bin/bundle.js HTTP/1.1" 200 -
192.168.1.107 - - [24/Aug/2017 08:37:04] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:04] "GET /favicon.ico HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:04] "GET /config.json HTTP/1.1" 200 -
192.168.1.107 - - [24/Aug/2017 08:37:05] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:05] "GET /var/www/nmap-logs/2017-08-24.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:05] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:05] "GET /var/www/nmap-logs/2017-08-23.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:05] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:05] "GET /var/www/nmap-logs/2017-08-22.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:05] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:05] "GET /var/www/nmap-logs/2017-08-21.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:05] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:05] "GET /var/www/nmap-logs/2017-08-20.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:05] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:05] "GET /var/www/nmap-logs/2017-08-19.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:05] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:05] "GET /var/www/nmap-logs/2017-08-18.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:37:05] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:37:05] "GET /logs/oldlogs HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:38:18] "GET / HTTP/1.1" 200 -
192.168.1.107 - - [24/Aug/2017 08:38:18] "GET /bin/bundle.js HTTP/1.1" 200 -
192.168.1.107 - - [24/Aug/2017 08:38:19] "GET /config.json HTTP/1.1" 200 -
192.168.1.107 - - [24/Aug/2017 08:38:19] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:38:19] "GET /var/www/nmap-logs/2017-08-24.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:38:19] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:38:19] "GET /var/www/nmap-logs/2017-08-23.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:38:19] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:38:19] "GET /var/www/nmap-logs/2017-08-22.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:38:19] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:38:19] "GET /var/www/nmap-logs/2017-08-21.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:38:19] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:38:19] "GET /var/www/nmap-logs/2017-08-20.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:38:19] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:38:19] "GET /var/www/nmap-logs/2017-08-19.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:38:20] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:38:20] "GET /var/www/nmap-logs/2017-08-18.xml HTTP/1.1" 404 -
192.168.1.107 - - [24/Aug/2017 08:38:20] code 404, message File not found
192.168.1.107 - - [24/Aug/2017 08:38:20] "GET /logs/oldlogs HTTP/1.1" 404 -

root@gdoor:/var/www/nmap-logs# 
root@gdoor:/var/www/nmap-logs# ls -l *xml
-rw-r--r-- 1 root root 126736 Aug 23 23:50 2017-08-23.xml
-rw-r--r-- 1 root root 120791 Aug 24 08:30 2017-08-24.xml
root@gdoor:/var/www/nmap-logs# 
root@gdoor:/var/www/nmap-logs# 
root@gdoor:/var/www/nmap-logs# ls -l /var/www/nmap-logs/2017-08-23.xml
-rw-r--r-- 1 root root 126736 Aug 23 23:50 /var/www/nmap-logs/2017-08-23.xml
root@gdoor:/var/www/nmap-logs# 

```

**Comment by phiresky** [comment]
`logFilesPath` in the config file needs to be relative to the server root. So in your example, it should probably be  `logFilesPath: "/nmap-logs/",`

Verify that you can access the log files from the browser via that path: <http://192.168.1.128:8000/nmap-logs/2017-08-23.xml>, if you can't do that you need to adjust the config file. Otherwise it's a bug elsewhere.

**Comment by jimkd1yv** [comment]
I got it to work.  I had to change logFilesPath:  "/"

It seems like python http.server must chroot of either the current directory, or the /var/www/nmap-files directory.

**Comment by phiresky** [comment]
Glad you got it working!

python http.server always served the directory you run it from as / I
think.

On Aug 24, 2017 7:18 PM, "jimkd1yv" <notifications@github.com> wrote:

I got it to work. I had to change logFilesPath: "/"

It seems like python http.server must chroot of either the current
directory, or the /var/www/nmap-files directory.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/phiresky/nmap-log-parse/issues/11#issuecomment-324699865>,
or mute the thread
